modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -1059,16 +1064,25 @@ public List<GridCacheClearAllRunnable<K, V>> splitClearLocally(boolean srv, bool
 
     /** {@inheritDoc} */
     @Override public void clear() throws IgniteCheckedException {
+        if (ctx.transactional() && ctx.grid().transactions().tx() != null)
```

Comment:
Code duplication. Check must be performed in a single place.

Reply:
Thank you, I will do it